cloudinit/stages.py

```diff
@@ -692,34 +705,77 @@ def _find_networking_config(self):
         return (self.distro.generate_fallback_config(),
                 NetworkConfigSource.fallback)
 
+    def update_event_enabled(self, event_source_type, scope=None):
+        # convert ds events to config
+        default_events = get_update_events_config(
+            self.datasource.default_update_events
+        )
+        config_events = self.cfg.get('updates', {})
+        allowed = get_enabled_events(
+            config_events, default_events
+        )
+
+        if not scope:
+            scopes = [allowed.keys()]
+        else:
+            scopes = [scope]
+        LOG.debug('Possible scopes for this event: %s', scopes)
+
+        for evt_scope in scopes:
+            if event_source_type in allowed.get(evt_scope, []):
+                LOG.debug('Event Allowed: scope=%s EventType=%s',
+                          evt_scope, event_source_type)
+                return True
+
+        LOG.debug('Event Denied: scopes=%s EventType=%s',
+                  scopes, event_source_type)
+        return False
```

Comment:
None of this references `self`: perhaps a candidate to move into `cloudinit.event` as a function?